client/helpers/scenes.ts

```diff
@@ -136,35 +117,47 @@ export async function createSceneAndSceneItems({
   onMsg = console.log,
   onProgress,
   parallelism,
-  polling,
+  polling = { intervalMs: PollIntervalMs, maxAttempts: MaxAttempts },
   verbose,
 }: CreateSceneAndSceneItemsReq): Promise<CreateSceneAndSceneItemsRes> {
   const scene = (
     await client.scenes.createScene({ createSceneRequest: createSceneReq() })
   ).data;
   const sceneId = scene.data.id;
+  const res = await createSceneItems({
+    client,
+    createSceneItemReqs,
+    failFast: failFast ?? false,
+    onProgress,
+    parallelism,
+    sceneId,
+  });
   const { a: queuedItems, b: errors } = partition(
-    (
-      await createSceneItems({
-        client,
-        createSceneItemReqs,
-        failFast: failFast ?? false,
-        onProgress,
-        parallelism,
-        sceneId,
-      })
-    ).queuedSceneItems,
-    (i: QueuedSceneItem) => !nullOrUndefined(i.job)
+    res.queuedSceneItems,
+    (i: QueuedSceneItem) => isQueuedJob(i.res)
   );
 
-  if (queuedItems.length === 0) return { errors, scene };
+  if (queuedItems.length === 0 || errors.length === res.leaves)
+    return { errors, scene };
```

Comment:
If all leaves had errors, bail out. The assemblies will succeed, but you won't see any parts so no use waiting for them.